app/src/main/java/eu/faircode/email/Core.java

```diff
@@ -1829,7 +1844,7 @@ static void downloadMessage(
             return;
 
         DB db = DB.getInstance(context);
-        EntityMessage message = db.message().getMessage(id);
+        DownloadMessage message = db.message().getDownloadMessageById(id);
```

Comment:
The question this raises is what's next?

Reply:
I agree it could be done in too many places and couldn't think of a more general solution to this yet.